src/llama_engine.cc

```diff
@@ -717,12 +739,17 @@ void LlamaEngine::HandleInferenceImpl(
     auto state = CreateInferenceState(si.ctx);
 
     // Queued task
-    si.q->runTaskInQueue([cb = std::move(callback), state, data, request_id]() {
+    si.q->runTaskInQueue([cb = std::move(callback), state, data, request_id, n_probs]() {
       state->task_id = state->llama.RequestCompletion(data, false, false, -1);
       while (state->llama.model_loaded_external) {
         TaskResult result = state->llama.NextResult(state->task_id);
         if (!result.error) {
-          std::string to_send = result.result_json["content"];
+          std::string to_send;
+          if (n_probs > 0){
```

Comment:
Can I verify my understanding about `n_probs`:

- From [llama.cpp server docs](https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md#post-completion-given-a-prompt-it-returns-the-predicted-completion) if `n_probs` > 0, resp contains probabilities of N tokens
- However, we don't seem to send the `content` back to the user in this case?
- Or: do we send both `content`, and also `completion_probabilities`? 

We should align with the conventions in llama.cpp's server, as much as possible

## Resources
- https://github.com/ggerganov/llama.cpp/issues/4088

Reply:
our implementation can return this form https://github.com/ggerganov/llama.cpp/issues/4088#issuecomment-1877510803, both content and list of (token - probs) for each token